sdk/python/kfp/components/_python_op.py

```diff
@@ -222,35 +222,57 @@ def _func_to_component_spec(func, extra_code='', base_image=_default_base_image,
     from collections import OrderedDict
     parameter_to_type_name = OrderedDict((input.name, str(input.type)) for input in component_spec.inputs)
 
-    input_args_parsing_code_lines =(
-        "    '{arg_name}': {arg_type}(sys.argv[{arg_idx}]),".format(
-            arg_name=name_type[0],
-            arg_type=name_type[1] if name_type[1] in ['int', 'float', 'bool'] else 'str',
-            arg_idx=idx + 1
+    arg_parse_code_lines = [
```

Comment:
nit: it's a bit harder to understand the code, better to add a comment with an example of generated code.

Reply:
Good idea. I'll add that to the PR description.